cmd/soroban-cli/src/config/identity/generate.rs

```diff
@@ -0,0 +1,33 @@
+use crate::config::locator;
+
+#[derive(thiserror::Error, Debug)]
+pub enum Error {
+    #[error("Unknown Error")]
+    Unknown,
+}
+
+#[derive(Debug, clap::Args)]
+pub struct Cmd {
+    /// Name of identity
+    pub name: String,
+    /// Optional seed to use when generating seed phrase
+    #[clap(long, short = 's')]
+    pub seed: Option<String>,
+
+    #[clap(flatten)]
+    pub config_locator: locator::Args,
+}
+
+impl Cmd {
+    pub fn run(&self) -> Result<(), Error> {
+        println!(
+            "Coming soon! This will generate a new identity named {} and a corresponding seed phrase from seed {:?}",
+            self.name,
+            self.seed.as_deref().unwrap_or("random")
+        );
+        if false {
+            return Err(Error::Unknown);
+        }
+        Ok(())
+    }
+}
```

Comment:
I'd vote for leaving out stuff that doesn't actually exist yet. The API may change when we implement it.

Reply:
For example, maybe we'd prefer to do `soroban config identity add mike --seed 5` instead.